### PR TITLE
Provide the total record count as well as the page of results.

### PR DESCRIPTION
**Pull request by tim-wilde-nimble**

### CODE REVIEW

File: TramsDataApi.Test/Controllers/AcademyTransferProjectsControllerTests.cs [hunk 1]
```diff
@@ -202,8 +203,10 @@ public void IndexAcademyTransferProject_ReturnsEmptyList_WhenThereAreNoAcademyTr
             var indexAcademyTransferProject = new Mock<IIndexAcademyTransferProjects>();
 
             indexAcademyTransferProject
-                .Setup(get => get.Execute(1))
-                .Returns(() => new List<AcademyTransferProjectSummaryResponse>());
+               .Setup(get => get.Execute(1))
+               .Returns(() =>
+                  Tuple.Create<IList<AcademyTransferProjectSummaryResponse>, int>(
```
Comment (jordan-beard) [inline TramsDataApi.Test/Controllers/AcademyTransferProjectsControllerTests.cs:208]: Small thing -
you can shorten this to - (new List<AcademyTransferProjectSummaryResponse>(), 0))

File: TramsDataApi/Gateways/IAcademyTransferProjectGateway.cs [hunk 2]
```diff
@@ -1,16 +1,29 @@
...
+      /// <param name="page">the page of results to return</param>
+      /// <returns>
+      ///    A <see cref="Tuple{IList, Int32}" /> containing the requested page of results and the total number of matching
+      ///    entries in the database
+      /// </returns>
+      /// <remarks>
+      ///    <para>Page size is hard-coded to ten</para>
+      ///    <para><paramref name="page" /> is 1-based, not zero-based (Page 1 is the first entry, not page zero)</para>
+      /// </remarks>
+      Tuple<IList<AcademyTransferProjects>, int> IndexAcademyTransferProjects(int page);
```
Comment (jordan-beard) [inline TramsDataApi/Gateways/IAcademyTransferProjectGateway.cs:26]: This can be shortened to (IList<AcademyTransferProjects>, int)

Comment (tim-wilde-nimble) [reply TramsDataApi/Gateways/IAcademyTransferProjectGateway.cs:26]: That would then give no hints about the return type to users of the interface, though.

File: TramsDataApi/UseCases/IndexAcademyTransferProjects.cs [hunk 3]
```diff
@@ -20,14 +20,42 @@ public class IndexAcademyTransferProjects : IIndexAcademyTransferProjects
...
         {
-            var listOfAcademyTransferProjects = _academyTransferProjectGateway.IndexAcademyTransferProjects(page);
+           Tuple<IList<AcademyTransferProjects>, int> listOfAcademyTransferProjects =
+              _academyTransferProjectGateway.IndexAcademyTransferProjects(page);
 
-            return listOfAcademyTransferProjects.ToList().Select(atp =>
-            {
-                return AcademyTransferProjectSummaryResponse(atp);
-            }).ToList();
+           return Tuple.Create<IList<AcademyTransferProjectSummaryResponse>, int>(
```
Comment (jordan-beard) [inline TramsDataApi/UseCases/IndexAcademyTransferProjects.cs:28]: Why can't you use the AcademyTransferProjectSummaryResponse constructor anymore?

File: TramsDataApi/Gateways/IAcademyTransferProjectGateway.cs (hunk 2 above)
Comment (jordan-beard) [reply TramsDataApi/Gateways/IAcademyTransferProjectGateway.cs:26]: Oh fair enough. I thought it was syntactic sugar but meant the same thing. Makes sense then.

File: TramsDataApi.Test/Controllers/AcademyTransferProjectsControllerTests.cs (hunk 1 above)
Comment (tim-wilde-nimble) [reply TramsDataApi.Test/Controllers/AcademyTransferProjectsControllerTests.cs:208]: Gave that (and a few variants) a go. That then doesn't conform to the return type, so won't compile :(

Comment (jordan-beard) [reply TramsDataApi.Test/Controllers/AcademyTransferProjectsControllerTests.cs:208]: Fair enough.

File: TramsDataApi/UseCases/IndexAcademyTransferProjects.cs (hunk 3 above)
Comment (tim-wilde-nimble) [reply TramsDataApi/UseCases/IndexAcademyTransferProjects.cs:28]: It does; two lines below this - I didn't change that code, just wrapped what was there into a tuple to include the total count.

Comment (tim-wilde-nimble) [reply TramsDataApi/UseCases/IndexAcademyTransferProjects.cs:28]: I see what you mean - Dom's change moved that code into the constructor. Missed that part of the merge conflicts.

Comment (tim-wilde-nimble) [reply TramsDataApi/UseCases/IndexAcademyTransferProjects.cs:28]: https://github.com/DFE-Digital/trams-data-api/commits/825e78590db4811740e2cf623b26e624282c4c56/TramsDataApi/UseCases/IndexAcademyTransferProjects.cs

Comment (tim-wilde-nimble) [reply TramsDataApi/UseCases/IndexAcademyTransferProjects.cs:28]: Updated to use the helper method - defintitely nicer to look at!

**Review by jordan-beard — APPROVED** [review]
